apache-http-client/src/main/java/com/joyent/http/signature/apache/httpclient/HttpSignatureAuthScheme.java

```diff
@@ -54,6 +68,13 @@
     private final ThreadLocalSigner signer;
 
     /**
+     * Map of credentials to cache object used for looking up cached signatures.
+     */
+    private ConcurrentMap<Credentials, HttpSignatureCache> signatureCacheMap =
```

Comment:
I'm having trouble understanding how many different Credentials there can be.  Isn't `HttpSignatureAuthScheme` a associated with a single `KeyPair`?   Are multiple credentials an accident of implementing `ContextAwareAuthScheme`, or a real supported use case?

If we expect this to really happen, I'm worried that this cache is unbounded in size.